core/services/vrf/listener_v2.go

```diff
@@ -271,19 +271,20 @@ func MaybeSubtractReservedLink(l logger.Logger, q pg.Q, startBalance *big.Int, c
 				   GROUP BY meta->>'SubId'`, chainID, subID)
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		l.Errorw("Could not get reserved link", "err", err)
-		return startBalance, err
+		return nil, err
 	}
 
 	if reservedLink != "" {
 		reservedLinkInt, success := big.NewInt(0).SetString(reservedLink, 10)
 		if !success {
 			l.Errorw("Error converting reserved link", "reservedLink", reservedLink)
-			return startBalance, errors.New("unable to convert returned link")
+			return nil, errors.New("unable to convert returned link")
 		}
-		// Subtract the reserved link
-		return startBalance.Sub(startBalance, reservedLinkInt), nil
+
+		return new(big.Int).Sub(startBalance, reservedLinkInt), nil
 	}
-	return startBalance, nil
+
+	return new(big.Int).Set(startBalance), nil
```

Comment:
An alternative would just be to make this function modify startBalance directly and only return "error", to me that matches the function name the best. Doesn't matter too much though

Reply:
Yeah, that's an alternative, though I think it's cleaner if we don't mutate the passed in args, it's simpler to reason about the code that way.